subprojects/docs/src/docs/userguide/tutorial/part5_gradle_caching.adoc

```diff
@@ -0,0 +1,238 @@
+- **On OS X / UNIX**: ~/.gradle/caches/
+
+A _cache key_ is used by Gradle to identity, quickly, which tasks and which inputs were used to create the cached results (outputs).
+If you use the `-i` argument you can see the _cache key_ used for each task input.
+
+Run a clean followed by a build to see this:
+[source]
+----
+$ ./gradlew :app:clean :app:build -i
+
+Initialized native services in: /Users/gradle-user/.gradle/native
+
+...
+
+> Task :app:compileJava FROM-CACHE
+Build cache key for task ':app:compileJava' is 7e6149c74419964bc38b387f6639a655
+Task ':app:compileJava' is not up-to-date because:
+  Output property 'destinationDirectory' file /Users/lkassovic/Documents/tutorial/app/build/classes/java/main has been removed.
+  Output property 'destinationDirectory' file /Users/lkassovic/Documents/tutorial/app/build/classes/java/main/com has been removed.
+  Output property 'destinationDirectory' file /Users/lkassovic/Documents/tutorial/app/build/classes/java/main/com/gradle has been removed.
+  Output property 'options.generatedSourceOutputDirectory' file /Users/lkassovic/Documents/tutorial/app/build/generated/sources/annotationProcessor/java/main has been removed.
+Loaded cache entry for task ':app:compileJava' with cache key 7e6149c74419964bc38b387f6639a655
+Resolve mutations for :app:processResources (Thread[Execution worker,5,main]) started.
+:app:processResources (Thread[Execution worker,5,main]) started.
+
+...
+----
+
+You can see the cache entries by looking in the cache directory:
+[source]
+----
+$ ls -ltr ~/.gradle/caches/build-cache-1/
+
+total 40
+-rw-r--r--  1 lkassovic  staff  3822 May 25 18:48 7e6149c74419964bc38b387f6639a655
+-rw-r--r--  1 lkassovic  staff  3373 May 25 18:48 5d1e0e63f2ba1b5e6797d4bf186d8fa6
+-rw-r--r--  1 lkassovic  staff  4217 May 25 18:48 e0ea15b80c7f67fdcd8b2902d1802da6
+-rw-r--r--  1 lkassovic  staff     0 May 25 18:48 gc.properties
+-rw-r--r--  1 lkassovic  staff    17 May 25 18:55 build-cache-1.lock
+----
+
+Here we see the cache key `7e6149c74419964bc38b387f6639a655`.
+[source]
+----
+$ ls -ltr ~/.gradle/caches/build-cache-1/7e6149c74419964bc38b387f6639a655
+
+total 40
+-rw-r--r--  1 lkassovic  staff  3822 May 25 18:48 7e6149c74419964bc38b387f6639a655
+-rw-r--r--  1 lkassovic  staff  3373 May 25 18:48 5d1e0e63f2ba1b5e6797d4bf186d8fa6
+-rw-r--r--  1 lkassovic  staff  4217 May 25 18:48 e0ea15b80c7f67fdcd8b2902d1802da6
+-rw-r--r--  1 lkassovic  staff     0 May 25 18:48 gc.properties
+-rw-r--r--  1 lkassovic  staff    17 May 25 18:55 build-cache-1.lock
+----
+`7e6149c74419964bc38b387f6639a655` is a TAR file.
+
+When the file is decompressed, we can see it contains our `App` class:
+
+image::tutorial/mac-tar-cache-exp.png[]
+
+Gradle will periodically clean-up the local cache directory by removing entries that have not been used recently to conserve disk space.
+
+== Step 5. Remote Caching
+The remote cache (called a Build Cache Node) is a shared build cache available via HTTP.
```

Comment:
I think the "Node" terminology is GE-specific and HTTP is too specific here.  I'd just say something like:

"In addition to the local cache present on a single developer's own machine, Gradle can make use of remote caches stored elsewhere, which can be used by multiple developers."